packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -916,8 +919,10 @@ class _DraggableScrollableSheetScrollPosition
       ..addListener(tick)
       ..animateWith(simulation).whenCompleteOrCancel(
         () {
-          _ballisticCancelCallback = null;
-          ballisticController.dispose();
+          if (_ballisticControllers.contains(ballisticController)) {
+            _ballisticControllers.remove(ballisticController);
+            ballisticController.dispose();
```

Comment:
This should be outside of the if statement.

Reply:
@dnfield
Thanks for reviewing this PR.
In some edge cases (the one in #101061, which is also the one in the test I added with this PR), code in `whenCompleteOrCancel` might be called after `_DraggableScrollableSheetScrollPosition.dispose`. If I kept this line outside the if statement it would throw an error about the AnimationController being already disposed.

To avoid this error, I put this line in the if statement and in `_DraggableScrollableSheetScrollPosition.dispose` I add a call to `_ballisticControllers.clear()` so if an AnimationController ends just after the call to `_DraggableScrollableSheetScrollPosition.dispose` it is no more in _ballisticControllers and  `ballisticController.dispose` won’t be called as it is in the if statement.  If the AnimationController ends before, it is still in `_ballisticControllers` and it will be disposed by this line (and removed from _ballisticControllers so we won't try to dispose it again in `_DraggableScrollableSheetScrollPosition.dispose`)

Reply:
Ahh ok, thanks, that makes sense.